daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilXMLLoader.scala

```diff
@@ -302,35 +306,30 @@ object DFDLCatalogResolver {
   def get = d.get
 }
 
-class Input(var pubId: String, var sysId: String, var inputStream: BufferedInputStream)
+class InputStreamLSInput(var pubId: String, var sysId: String, inputStream: InputStream)
   extends LSInput {
 
   var myBaseURI: String = null
 
+  def getBaseURI = myBaseURI
   def getPublicId = pubId
+  def getSystemId = sysId
+
+  def setBaseURI(baseURI: String) = myBaseURI = baseURI
   def setPublicId(publicId: String) = pubId = publicId
-  def getBaseURI = myBaseURI
-  def getByteStream = null
+  def setSystemId(systemId: String) = sysId = systemId
+
+  def getByteStream = inputStream
   def getCertifiedText = false
   def getCharacterStream = null
   def getEncoding = null
-  def getStringData = {
-    this.synchronized {
-      val input: Array[Byte] = new Array[Byte](inputStream.available())
-      inputStream.read(input)
-      val contents = new String(input)
-      contents
-    }
-  }
-  def setBaseURI(baseURI: String) = myBaseURI = baseURI
+  def getStringData = null
```

Comment:
Can you add a comment explaining why this is important? Especially considering we do not have tests to catch the problems this solves.

Reply:
Yep, will do.